dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ProviderConfig.java

```diff
@@ -157,7 +158,7 @@
 
     @Deprecated
     public void setProtocol(String protocol) {
-        this.protocols = Arrays.asList(new ProtocolConfig[]{new ProtocolConfig(protocol)});
+        this.protocols = new ArrayList<>(Arrays.asList(new ProtocolConfig[]{new ProtocolConfig(protocol)}));

```

Comment:
This can be simplified as `new ArrayList<>(Arrays.asList(new ProtocolConfig(protocol)))`